WordPress/Classes/Services/MediaThumbnailCoordinator.swift

```diff
@@ -53,11 +41,18 @@ class MediaThumbnailCoordinator: NSObject {
             DispatchQueue.main.async {
                 onCompletion(image, nil)
             }
-        }, onError: { (error) in
+        }
+        let failure: (Error?) -> Void = { (error) in
             DispatchQueue.main.async {
                 onCompletion(nil, error)
             }
-        })
+        }
+
+        coreDataStack.performAndSave { context in
+            let mediaThumbnailService = MediaThumbnailService(managedObjectContext: context)
+            mediaThumbnailService.exportQueue = self.queue
+            mediaThumbnailService.thumbnailURL(forMedia: media, preferredSize: size, onCompletion: success, onError: failure)
```

Comment:
Was there a particular threading or memory reason for extracting `success` and `failure` as opposed to define them as closure parameters in the method call, or just style?

Reply:
Just style. I wanted to minimize the diff mainly. 😄

Reply:
Sounds good. Merge at will 👍